src/core/activityContent/validation.ts

```diff
@@ -40,386 +42,398 @@ const SUPPORTED_IMAGE_MEDIA_TYPES = [
-  }
-  if (location["units"]) {
-    if (!isString(location["units"])) return false;
-    if (
-      location["units"] !== "cm" &&
-      location["units"] !== "feet" &&
-      location["units"] !== "inches" &&
-      location["units"] !== "km" &&
-      location["units"] !== "m" &&
-      location["units"] !== "miles"
-    )
-      return false;
+/**
+ * requireGetSupportedContentAttachments iterates over the provided list and returns
+ * the valid attachments. If the passed in obj is empty, returns an empty array.
+ * If there are attachments but none of them are valid, it throws an error.
+ *
+ * @param obj - an array of unknown object types to check for validity
+ * @returns an array of all valid ActivityContentAttachments
+ * @throws InvalidActivityContentError if there are no valid attachments.
+ */
+export const requireGetSupportedContentAttachments = (obj: Array<unknown>): Array<ActivityContentAttachment> => {
+  if (!obj.length) return [];
+  const validTypeAttachments = obj.filter((attachment) => {
+    return !!toActivityContentAttachment(attachment);
+  });
+  if (!validTypeAttachments.length) {
+    for (const attachment of obj) {
+      requireToActivityContentAttachment(attachment);
+    }
   }
-
-  return true;
+  return validTypeAttachments as Array<ActivityContentAttachment>;
 };
 
 /**
- * isValidActivityContentNote() validates that a given object meets all
+ * requireValidActivityContentNote() validates that a given object meets all
  * necessary specification requirements for an ActivityContentNote object,
  * include string field formats and other logical validations. Note that this
  * function **does not** attempt to fetch any attachments to validate that their
  * hashes are correct.
  *
  * @param obj - The object to check
- * @returns A boolean indicating whether or not the object is a valid ActivityContentNote
+ * @returns true if valid
+ * @throws InvalidActivityContentError if not valid, with a message
  */
-export const isValidActivityContentNote = (obj: unknown): boolean => {
-  if (!isActivityContentNote(obj)) return false;
-
-  if (obj["published"] && !isValidPublishedField(obj["published"])) return false;
-  if (obj["attachment"]) {
-    for (const attachment of obj["attachment"]) {
-      if (!isValidActivityContentAttachment(attachment)) return false;
+export const requireValidActivityContentNote = (obj: unknown): boolean => {
+  if (requireIsActivityContentNoteType(obj)) {
+    if (obj["tag"]) requireIsActivityContentTag(obj["tag"]);
+
+    if (obj["published"] && !isValidPublishedField(obj["published"]))
+      throw new InvalidActivityContentError("ActivityContentNote published is invalid");
+    if (obj["attachment"]) {
+      const attachments = requireToArray(obj["attachment"], "ActivityContentNote attachment");
+      requireGetSupportedContentAttachments(attachments);
     }
+    if (obj["location"]) requireIsActivityContentLocation(obj["location"]);
```

Comment:
Are multiple locations not supported for notes anymore? If so that should be noted in changelog.

Reply:
No, we discussed it and decided while you were out.  I'll put it in the Changelog.